usr/share/rear/prep/GNU/Linux/210_include_dhclient.sh

```diff
@@ -72,6 +72,22 @@ function dhcp_client_is_active() {
         done
     fi
 
+    # Strategy 3: Check if Network Manager has been used for DHCP client setup
+    if type nmcli &>/dev/null; then
+        local conn_name
+        while read -r conn_name; do
+            if [[ -n "$conn_name" ]]; then
+                if [[ "$(nmcli --get-values ipv4.method connection show "$conn_name")" == "auto" ]]; then
```

Comment:
When I try this on my Debian Stretch, I got following error:
```
~# nmcli --get-values ipv4.method connection show "$conn_name"
Error: Option '-get-values' is unknown, try 'nmcli -help'.
```

This will probably mean that older versions on `nmcli` had different synopsis. I'd recommend to move this code to be executed only on distros with support such syntax.

V.

Reply:
Thanks for testing that quickly! I had looked up nmcli on Ubuntu 20.04 and 18.04 where the above syntax is present. If it's not on other systems I agree, we should take care of that, too. I'll come up with an additional commit.

Reply:
FYI: Network Manager got [this commit on May 25, 2017](https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/commit/7c24a2cb39805d413b8d07f0b2e5dfa858b8eec2):

> cli: allow the GNU option format
  $ nmcli --fields=all c
  Error: Option '-fields=all' is unknown, try 'nmcli -help'.
>
> What a shame. Let's fix this.

I suppose older versions of nmcli only appear on systems where other methods of DHCP auto-detection did work. Otherwise we would probably have already seen it. As this PR's method is the last in a chain, the older detection methods have priority and this one should not interfere with them.

Reply:
I think code like
```
if [[ "$( COMMAND )" == "EXPECTED_STDOUT" ]] ; then echo OK ; fi
```
is perfectly fine because it behaves safe when COMMAND fails like
```
# if [[ "$( cat qqq )" == "something" ]] ; then echo OK ; fi
cat: qqq: No such file or directory
```
If one does not want to see the stderr message one could even use
```
if [[ "$( cat qqq 2>/dev/null )" == "something" ]] ; then echo OK ; fi
```

The function `dhcp_client_is_active` is only called once
for `Auto-enabling DHCP on the rescue system`.

I think is is neither expected nor needed that ReaR
could do all stuff fully automated in the right way.

When the user needs DHCP in the recovery system he can and should
explicitly specify that in his local.conf via `USE_DHCLIENT="yes"`